content/en/docs/concepts/workloads/controllers/job.md

```diff
@@ -807,6 +807,16 @@ These are some requirements and semantics of the API:
   - `Count`: use to indicate that the Pod should be handled in the default way.
      The counter towards the `.spec.backoffLimit` should be incremented.
 
+{{< note >}}
+Since the 1.27 version Kubelet transitions deleted pods to a terminal phase
+(see: [Pod Phase](/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase)).
+This solves the known issue for using pod failure policy in 1.26.
+In the previous version a pod could get stuck in the `Pending` phase if deleted
+while in that phase. This is because Job controller delays removal of the pod
+finalizer (hence blocking the pod deletion from the API server) until the pod is
+transitioned into a terminal phase.
```

Comment:
This is content for a release note. We generally don't describe old behavior in documentation.

Maybe you can instead add examples of why a pod would be deleted? Like explicit deletion, eviction API, etc.

Reply:
Updated the release notes under the PR: https://github.com/kubernetes/kubernetes/pull/115331.

I guess saying why a pod would be deleted doesn't really belong to jobs page either. Maybe on the other hand I will just describe that every pod ends up in a terminal phase and link to Pod lifecycle.

Reply:
Deleted the description of the old behavior, but still keep the note to refer to the Pod lifecycle.